broker.go

```diff
@@ -38,23 +20,41 @@ const (
 	DefaultFailureSleep = 100 * time.Millisecond
 )
 
+// Broker represents an InfluxDB specific messaging broker.
+type Broker struct {
+	*messaging.Broker
+
+	done chan struct{}
+
+	// send CQ processing requests to the same data node
+	// currentCQProcessingNode *messaging.Replica // FIX(benbjohnson)
+
+	// variables to control when to trigger processing and when to timeout
+	TriggerInterval     time.Duration
+	TriggerTimeout      time.Duration
+	TriggerFailurePause time.Duration
+}
+
 // NewBroker returns a new instance of a Broker with default values.
 func NewBroker() *Broker {
-	b := &Broker{
+	return &Broker{
+		Broker:              messaging.NewBroker(),
 		TriggerInterval:     5 * time.Second,
 		TriggerTimeout:      20 * time.Second,
 		TriggerFailurePause: 1 * time.Second,
 	}
-	b.Broker = messaging.NewBroker()
-	return b
 }
 
 // RunContinuousQueryLoop starts running continuous queries on a background goroutine.
 func (b *Broker) RunContinuousQueryLoop() {
-	b.done = make(chan struct{})
-	go b.continuousQueryLoop(b.done)
+	// FIX(benbjohnson)
+	// b.done = make(chan struct{})
```

Comment:
The brokers now only track the highest index received by any data node so we aren't tracking the actual data node URLs themselves. I realized that was an issue once I got to CQs and remembered that they need cluster info to work.

I talked to Paul about this and we decided to send the data node URLs up to the broker. There's a periodic ping that occurs from the client so I can add it there. I disabled the CQs temporarily because it was more important to get messaging stable.